***Data Cleaning***

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib as pt 
import seaborn as sns 


In [2]:
df = pd.read_csv('college_student_placement_dataset.csv')

In [3]:
df.shape

(10000, 10)

In [4]:
df.isnull().sum()

College_ID                0
IQ                        0
Prev_Sem_Result           0
CGPA                      0
Academic_Performance      0
Internship_Experience     0
Extra_Curricular_Score    0
Communication_Skills      0
Projects_Completed        0
Placement                 0
dtype: int64

In [5]:
df.duplicated().sum()

np.int64(0)

In [6]:
df.head()

College_ID   IQ  Prev_Sem_Result  CGPA  Academic_Performance  \
0    CLG0030  107             6.61  6.28                     8   
1    CLG0061   97             5.52  5.37                     8   
2    CLG0036  109             5.36  5.83                     9   
3    CLG0055  122             5.47  5.75                     6   
4    CLG0004   96             7.91  7.69                     7   

  Internship_Experience  Extra_Curricular_Score  Communication_Skills  \
0                    No                       8                     8   
1                    No                       7                     8   
2                    No                       3                     1   
3                   Yes                       1                     6   
4                    No                       8                    10   

   Projects_Completed Placement  
0                   4        No  
1                   0        No  
2                   1        No  
3                   1        No  
4                   2        No

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 10 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   College_ID              10000 non-null  object 
 1   IQ                      10000 non-null  int64  
 2   Prev_Sem_Result         10000 non-null  float64
 3   CGPA                    10000 non-null  float64
 4   Academic_Performance    10000 non-null  int64  
 5   Internship_Experience   10000 non-null  object 
 6   Extra_Curricular_Score  10000 non-null  int64  
 7   Communication_Skills    10000 non-null  int64  
 8   Projects_Completed      10000 non-null  int64  
 9   Placement               10000 non-null  object 
dtypes: float64(2), int64(5), object(3)
memory usage: 781.4+ KB


In [8]:
df.dtypes

College_ID                 object
IQ                          int64
Prev_Sem_Result           float64
CGPA                      float64
Academic_Performance        int64
Internship_Experience      object
Extra_Curricular_Score      int64
Communication_Skills        int64
Projects_Completed          int64
Placement                  object
dtype: object

In [9]:
df.describe()

IQ  Prev_Sem_Result          CGPA  Academic_Performance  \
count  10000.000000     10000.000000  10000.000000          10000.000000   
mean      99.471800         7.535673      7.532379              5.546400   
std       15.053101         1.447519      1.470141              2.873477   
min       41.000000         5.000000      4.540000              1.000000   
25%       89.000000         6.290000      6.290000              3.000000   
50%       99.000000         7.560000      7.550000              6.000000   
75%      110.000000         8.790000      8.770000              8.000000   
max      158.000000        10.000000     10.460000             10.000000   

       Extra_Curricular_Score  Communication_Skills  Projects_Completed  
count            10000.000000          10000.000000        10000.000000  
mean                 4.970900              5.561800            2.513400  
std                  3.160103              2.900866            1.715959  
min                  0.000000              1.000000            0.000000  
25%                  2.000000              3.000000            1.000000  
50%                  5.000000              6.000000            3.000000  
75%                  8.000000              8.000000            4.000000  
max                 10.000000             10.000000            5.000000

In [10]:
df['CGPA'].describe()

count    10000.000000
mean         7.532379
std          1.470141
min          4.540000
25%          6.290000
50%          7.550000
75%          8.770000
max         10.460000
Name: CGPA, dtype: float64

In [11]:
df['CGPA'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 10000 entries, 0 to 9999
Series name: CGPA
Non-Null Count  Dtype  
--------------  -----  
10000 non-null  float64
dtypes: float64(1)
memory usage: 78.3 KB


In [12]:
df['Internship_Experience'].describe()


count     10000
unique        2
top          No
freq       6036
Name: Internship_Experience, dtype: object

In [13]:
df['Internship_Experience'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 10000 entries, 0 to 9999
Series name: Internship_Experience
Non-Null Count  Dtype 
--------------  ----- 
10000 non-null  object
dtypes: object(1)
memory usage: 78.3+ KB


In [14]:
df['Projects_Completed'].describe()

count    10000.000000
mean         2.513400
std          1.715959
min          0.000000
25%          1.000000
50%          3.000000
75%          4.000000
max          5.000000
Name: Projects_Completed, dtype: float64

In [15]:
df['Projects_Completed'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 10000 entries, 0 to 9999
Series name: Projects_Completed
Non-Null Count  Dtype
--------------  -----
10000 non-null  int64
dtypes: int64(1)
memory usage: 78.3 KB


In [16]:
df['Projects_Completed'].value_counts()

Projects_Completed
5    1702
4    1693
0    1688
2    1681
3    1627
1    1609
Name: count, dtype: int64

In [17]:
df['Extra_Curricular_Score'].describe()

count    10000.000000
mean         4.970900
std          3.160103
min          0.000000
25%          2.000000
50%          5.000000
75%          8.000000
max         10.000000
Name: Extra_Curricular_Score, dtype: float64

In [18]:
df['Communication_Skills'].describe()

count    10000.000000
mean         5.561800
std          2.900866
min          1.000000
25%          3.000000
50%          6.000000
75%          8.000000
max         10.000000
Name: Communication_Skills, dtype: float64

In [19]:
df['Communication_Skills'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 10000 entries, 0 to 9999
Series name: Communication_Skills
Non-Null Count  Dtype
--------------  -----
10000 non-null  int64
dtypes: int64(1)
memory usage: 78.3 KB


In [20]:
df['Academic_Performance'].describe()

count    10000.000000
mean         5.546400
std          2.873477
min          1.000000
25%          3.000000
50%          6.000000
75%          8.000000
max         10.000000
Name: Academic_Performance, dtype: float64

In [21]:
df['Prev_Sem_Result'].describe()

count    10000.000000
mean         7.535673
std          1.447519
min          5.000000
25%          6.290000
50%          7.560000
75%          8.790000
max         10.000000
Name: Prev_Sem_Result, dtype: float64

In [22]:
df['IQ'].describe()

count    10000.000000
mean        99.471800
std         15.053101
min         41.000000
25%         89.000000
50%         99.000000
75%        110.000000
max        158.000000
Name: IQ, dtype: float64

In [23]:
df['Placement'].describe()

count     10000
unique        2
top          No
freq       8341
Name: Placement, dtype: object

In [24]:
df['Placement'].value_counts()

Placement
No     8341
Yes    1659
Name: count, dtype: int64

***Summary***
- Columns = 10 (one of the column is college_id -> which is not so useful)
- Rows = 10000
- No NULL value in dataset.
- No duplicate rows 

***Table -> CGPA***
- Numerical column -> float64
- No outliers or unusualness in CGPA column

***Table -> Internship_Experience***
- Categorical column -> object
- No outliers or unusualness in Internship_Experience column

***Table -> Projects_Completed***
- Numerical column -> int64
- No outliers or unusualness in Projects

***Table -> Extra_Curricular_Score***
- Numerical column -> float64
- out of 10

***Table -> Communication_Skills***
- Numerical column -> float64
- out of 10
- No outliers or unusualness in Workshops/Certifications

***Table -> Academic_Performance***
- Numerical column -> float64
- No outliers or unusualness in Academic_Performance

***Table -> Prev_Sem_Result***
- Numerical column -> float64

***Table -> IQ***
- Numerical column -> float64
- min -> 41
- max -> 158

***Table -> Placement***
- Categorical colum -> Yes/No
- No -> 8341
- Yes -> 1659
- only small percentage of students got placement. All others are rejected .That is Data is unbalanced and it naturally tends rejected case
